### ISSUE TITLE
Normals inverted on imported models

### ISSUE THREAD

**Issue by ee0pdt** [godotengine/godot#39936]
**Godot version:**
3.2.2

**OS/device including version:**
macOS 10.14.6 Mojave
2017 MacBook Pro 13", 2.3Ghz Core i5
Intel Iris Plus 640
GLES3 Project

**Issue description:**
On import models have inverted normals (see direction of light in screenshot)
![image](https://user-images.githubusercontent.com/6094917/85984458-20f61600-b9e1-11ea-97bd-efd8b062e33c.png)

**Steps to reproduce:**
Create model in Blender
Export as .glb
Import into Godot
Normals are inverted

**Minimal reproduction project:**
[Normals Bug.zip](https://github.com/godotengine/godot/files/4844317/Normals.Bug.zip)

**Comment by Calinou** [comment]
Which Blender version and exporter did you use to export the `.glb` file?

**Comment by mbrlabs** [comment]
I can't reproduce this in 3.2.2 with your provided project on Manajro Linux.

**Comment by ee0pdt** [comment]
@Calinou 2.83.0 - But it's not just Blender, SketchFab models have the same issue. Also the same issue is present for .fbx imports

![image](https://user-images.githubusercontent.com/6094917/86019899-1fe3d980-ba1f-11ea-8620-a47fa7a0cc82.png)

@mbrlabs - I suspect this is a Mac only issue as no-one else seems to have reported issues. I think there are far fewer Mac users of Godot than Linux/Windows (just a guess there)

**Comment by ee0pdt** [comment]
Here is an updated example Project with FBX import and a standard CSG Box for comparision

[Normals Bug 2.zip](https://github.com/godotengine/godot/files/4846209/Normals.Bug.2.zip)

**Comment by ee0pdt** [comment]
Update: Issue is not present on .obj files

**Comment by ee0pdt** [comment]
I cannot replicate this on a different Mac with Catalina and a Geforce GPU. Wondering if this is related to the graphics chip or the OS, will try to dig further

**Comment by ee0pdt** [comment]
@Calinou I've determined that this is more likely a rendering bug as it only affects imported models when rendering on Intel Iris graphics. The same Mac connected to an eGPU with a Radeon Vega card does not display the issue. Should I close this ticket and make a new one? Or change this ticket to reflect my findings?

**Comment by akien-mga** [comment]
Might be related to #39975.

Can you reproduce it with GLES2 too, or is it a GLES3-only bug?

**Comment by snougo** [comment]
I can't reproduce this issue in 3.2.2 with your provided project on MacOS 10.13.6 when using Intel Iris Graphics, my device is a 2013 late MacBook Pro 13-inch, pretty old

here is screenshot
<a data-flickr-embed="true" href="https://www.flickr.com/photos/189186320@N08/50067577053/in/dateposted-public/" title="屏幕快照 2020-07-02 18.01.40"><img src="https://live.staticflickr.com/65535/50067577053_aa23d24c4f_c.jpg" width="800" height="478" alt="屏幕快照 2020-07-02 18.01.40"></a>

**Comment by ee0pdt** [comment]
Thanks @snougo. What OS are you running?

@akien-mga Not tried GLES 2 yet, hopefully get a chance after work today

**Comment by ee0pdt** [comment]
@akien-mga GLES 2 has the same rendering issue

**Comment by jonathanhirz** [comment]
Been seeing this same issue on my Mac Mini (2018, Intel UHD Graphics 630). I attached screenshots of your two example projects. CSG objects created in Godot are lit correctly, but if I try to import from blender the lighting seems backwards. Hope this can get fixed soon, there ARE Godot users with Macs out there (dozens of us!) =]

![Screen Shot 2021-01-02 at 8 18 37 PM](https://user-images.githubusercontent.com/1631578/103471688-08700b00-4d38-11eb-921d-2f918072b7b6.png)
![Screen Shot 2021-01-02 at 8 17 48 PM](https://user-images.githubusercontent.com/1631578/103471689-0c039200-4d38-11eb-9a04-a802282d8890.png)

**Comment by Calinou** [comment]
@jonathanhirz As a workaround, you can switch to the GLES2 renderer in the top-right corner of the editor.

**Comment by jonathanhirz** [comment]
> @jonathanhirz As a workaround, you can switch to the GLES2 renderer in the top-right corner of the editor.

ah nice, thanks I'll give that a try going forward. I did try it on the examples here and it seemed to light the correct side.

**Comment by jonathanhirz** [comment]
![Screen Shot 2021-04-19 at 11 48 05 AM](https://user-images.githubusercontent.com/1631578/115287979-bd6edd80-a105-11eb-9e1b-7e7606597e5f.png)
Just wanted to mention that I am still seeing this issue on my Mac throughout the RCs that have been released. It seems like the only cubes that are lit on the correct side are the CSG box created in Godot, and an OBJ imported from blender. Other formats (gtfl and dae) are lit backwards. I'm hoping that the 4.0 renderer will make this bug obsolete :)

**Comment by Polirecyliente** [comment]
I can reproduce and confirm this bug in a Mac. I inverted the normals in Blender so it looks good in GLES3 (so the face orientation looks red in Blender, while the lighting looks correct in Godot v3.4).

The Mac in which I reproduced this bug has the following metadata:
- macOS Big Sur, version 11.6.1
- MacBook Pro (13-inch, 2020, Four Thunderbolt 3 ports)
- Processor 2.3 GHz Quad-Core Intel Core i7
- Memory 16 GB 3733 MHz LPDDR4X
- Graphics Intel Iris Plus Graphics 1536 MB

Thank you.

**Comment by tavurth** [comment]
Just ran into this issue, after finally figuring out how to import animations in 3.4.2.

Are there any other workarounds which allow the use of GLES3? 

This issue also means at the current time there seems to be no way to import **animated** 3d models (which render correctly) into Godot for all versions of Mac OS. All animation must be done via Godot using the OBJ importer which will not work for most peoples workflow.

Perhaps someone could point me to the general area of the codebase so I can have a dig around there to see if I can narrow down the issue?

**Comment by tavurth** [comment]
After some digging I have found a workaround, and perhaps the issue itself:

I reImported my model, and asked Godot to save the materials as a separate file:
<img width="300" alt="Screenshot 2022-01-19 at 00 08 03" src="https://user-images.githubusercontent.com/100964/150019090-5f562111-d7ce-4d0c-bdf2-b3753c2ae637.png">

I then added the material to the animated model by `clearing inheritance` and I saw that the `cull mode` option was checked.
<img width="300" alt="Screenshot 2022-01-19 at 00 06 52" src="https://user-images.githubusercontent.com/100964/150018882-e43c2394-b926-4b91-b590-257f2278daa1.png">

Which results in the following image (same as before)
<img width="500" alt="Screenshot 2022-01-19 at 00 07 03" src="https://user-images.githubusercontent.com/100964/150018947-8ed9fda3-18ef-4a3e-82a6-f74fe92ca870.png">

After reverting the `cull mode` to `back`, the issue seems to be resolved:

<img width="500" alt="Screenshot 2022-01-19 at 00 07 18" src="https://user-images.githubusercontent.com/100964/150019320-64dd0f0c-f3be-401f-8bd6-cd7ccac2fd20.png">

@Calinou perhaps as a good temporary solution we should prevent the `glb` importer from writing to the `cull mode` of materials?

**Comment by Calinou** [comment]
> @Calinou perhaps as a good solution we should prevent the glb importer from writing to the cull mode of materials?

Indeed, the Back cull mode should be used for glTF materials, but this may be contrary to the glTF specification. See also https://github.com/godotengine/godot/pull/53642.

**Comment by jonathanhirz** [comment]
Just wanted to confirm, back face culling seems to "fix" this issue I was seeing too on my Mac Mini (specs above). I did a quick test on my box project, and if I edited the material to use back face culling, the lighting looked right. Also, I tried setting this on in blender on the material file, and the imported model looked good in Godot. Not sure how this fits in to default settings or what all this changes, but at least there's a way to get lighting right on these machines. Thanks!
